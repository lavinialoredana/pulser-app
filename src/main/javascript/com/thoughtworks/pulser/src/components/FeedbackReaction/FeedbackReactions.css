.Feedback-reaction-container {
    width: calc(100% / 4);
    height: 400px;
    margin: 0 3rem;
}

.Feedback-reaction-image {
    height: 200px;
    width: 200px;
    margin-top: 2rem;
    display: flex;
    flex-direction: column;
}