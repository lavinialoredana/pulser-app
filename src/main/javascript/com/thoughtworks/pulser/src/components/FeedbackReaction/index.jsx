import "./FeedbackReactions.css";

const FeedbackReaction = ({
  reactionName,
  reactionImage,
  reactionAlt,
  reactionKey,
  onReactionClick,
  isReactionPressed,
}) => {
  const onClick = (event) => {
    onReactionClick(isReactionPressed(reactionKey) ? " " : reactionKey )
    
    // isReactionPressed(reactionKey)? onReactionClick(""): onReactionClick(reactionKey);
  };

  return (
    <div
      className={`Feedback-reaction-component ${
        isReactionPressed(reactionKey) ? "Is-clicked" : ""
      } `}
    >
      <h3 data-testid="feedBackReaction-title"> {reactionName}</h3>
      <img
        src={reactionImage}
        alt={reactionAlt}
        className="Feedback-reaction-image"
        onClick={onClick}
      />
    </div>
  );
};

export default FeedbackReaction;
