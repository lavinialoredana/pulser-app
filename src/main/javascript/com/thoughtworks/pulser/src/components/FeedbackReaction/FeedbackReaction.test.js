import { cleanup, render, screen } from "@testing-library/react";
import FeedbackReaction from "./index.jsx";

describe("FeedbackReaction Component", () => {
  afterEach(() => {
    cleanup();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(
      <FeedbackReaction
        reactionName="Awesome"
        reactionImage="awesome_face"
        reactionAlt="awesome_face"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should contain h3 title", () => {
    render(<FeedbackReaction />);

    const titleElement = screen.getByTestId("feedBackReaction-title");

    expect(titleElement).toBeInTheDocument();
  });

  it("should render the title", () => {
    render(<FeedbackReaction reactionName="Awesome" />);

    const titleText = screen.getByText("Awesome");

    expect(titleText).toBeInTheDocument();
  });

  it("displays an awesome face", () => {
    render(
      <FeedbackReaction
        reactionImage="awesome_face"
        reactionAlt="awesome_face"
      />
    );
    const awesomeImage = screen.getByAltText("awesome_face");
    
    expect(awesomeImage.src).toContain("awesome_face");
  });
});
