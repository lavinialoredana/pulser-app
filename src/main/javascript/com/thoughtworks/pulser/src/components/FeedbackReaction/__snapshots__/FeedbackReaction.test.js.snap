// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackReaction Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Feedback-reaction-component  "
    data-testid="feedback-reaction"
  >
    <h3
      data-testid="feedBackReaction-title"
    >
       Awesome
    </h3>
    <img
      alt="awesome_face"
      class="Feedback-reaction-image"
      src="awesome_face"
    />
  </div>
</DocumentFragment>
`;
